keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,891 @@
+    "resnet-rs-350-i320.h5": "7ccb858cc738305e8ceb3c0140bee393",
+    "resnet-rs-350-i320_notop.h5": "ab0c1f9079d2f85a9facbd2c88aa6079",
+    "resnet-rs-420-i320.h5": "ae0eb9bed39e64fc8d7e0db4018dc7e8",
+    "resnet-rs-420-i320_notop.h5": "fe6217c32be8305b1889657172b98884",
+    "resnet-rs-50-i160.h5": "69d9d925319f00a8bdd4af23c04e4102",
+    "resnet-rs-50-i160_notop.h5": "90daa68cd26c95aa6c5d25451e095529",
+}
+
+DEPTH_TO_WEIGHT_VARIANTS = {
+    50: [160],
+    101: [160, 192],
+    152: [192, 224, 256],
+    200: [256],
+    270: [256],
+    350: [256, 320],
+    420: [320],
+}
+BLOCK_ARGS = {
+    50: [
+        {"input_filters": 64, "num_repeats": 3},
+        {"input_filters": 128, "num_repeats": 4},
+        {"input_filters": 256, "num_repeats": 6},
+        {"input_filters": 512, "num_repeats": 3},
+    ],
+    101: [
+        {"input_filters": 64, "num_repeats": 3},
+        {"input_filters": 128, "num_repeats": 4},
+        {"input_filters": 256, "num_repeats": 23},
+        {"input_filters": 512, "num_repeats": 3},
+    ],
+    152: [
+        {"input_filters": 64, "num_repeats": 3},
+        {"input_filters": 128, "num_repeats": 8},
+        {"input_filters": 256, "num_repeats": 36},
+        {"input_filters": 512, "num_repeats": 3},
+    ],
+    200: [
+        {"input_filters": 64, "num_repeats": 3},
+        {"input_filters": 128, "num_repeats": 24},
+        {"input_filters": 256, "num_repeats": 36},
+        {"input_filters": 512, "num_repeats": 3},
+    ],
+    270: [
+        {"input_filters": 64, "num_repeats": 4},
+        {"input_filters": 128, "num_repeats": 29},
+        {"input_filters": 256, "num_repeats": 53},
+        {"input_filters": 512, "num_repeats": 4},
+    ],
+    350: [
+        {"input_filters": 64, "num_repeats": 4},
+        {"input_filters": 128, "num_repeats": 36},
+        {"input_filters": 256, "num_repeats": 72},
+        {"input_filters": 512, "num_repeats": 4},
+    ],
+    420: [
+        {"input_filters": 64, "num_repeats": 4},
+        {"input_filters": 128, "num_repeats": 44},
+        {"input_filters": 256, "num_repeats": 87},
+        {"input_filters": 512, "num_repeats": 4},
+    ],
+}
+CONV_KERNEL_INITIALIZER = {
+    "class_name": "VarianceScaling",
+    "config": {
+        "scale": 2.0,
+        "mode": "fan_out",
+        "distribution": "truncated_normal"
+    }
+}
+
+def Conv2DFixedPadding(filters, kernel_size, strides, name):
+    """Conv2D block with fixed padding.
+    """
+    def apply(inputs):
+        if strides > 1:
+            inputs = fixed_padding(inputs, kernel_size)
+        return layers.Conv2D(
+            filters=filters,
+            kernel_size=kernel_size,
+            strides=strides,
+            padding="same" if strides == 1 else "valid",
+            use_bias=False,
+            kernel_initializer=CONV_KERNEL_INITIALIZER,
+            name=name,
+        )(inputs)
+    return apply
+
+def STEM(bn_momentum: float, bn_epsilon: float, activation: str):
```

Comment:
Inconsistent treatment of `name`: either you use it everywhere (preferred) or nowhere

Reply:
Added name.